Get all tweets of user

Noisyboy-9 opened the pull request:
In this pr, we are trying to get the list of all tweets for a specific user. The route for doing so is /users/student_id/tweets.

Noisyboy-9 committed:
Remove unnecessary .gitkeep from folders

Noisyboy-9 committed:
Rename files to obate project naming conventions

Noisyboy-9 committed:
Add user.route.js

Noisyboy-9 committed:
Add users query for geting all users and getting user by id

Noisyboy-9 committed:
Finish users controller for getting list of all users

Noisyboy-9 committed:
Add getting list of all users to list of routes

Noisyboy-9 committed:
Change usersController.getAllUsers() res to have predefined structure

Noisyboy-9 committed:
Add auth check middleware

Noisyboy-9 committed:
Change get all users query to just get id,name,student_id,avatar of each user

Noisyboy-9 committed:
Finish feature

Noisyboy-9 committed:
Finish feature

arefehkompani commented on src/db/queries/tarin/update.js:
```diff
@@ -1,24 +1,24 @@
-const User = require('../../models/user');
+const user = require('../../models/user');
```
based on your previous comments I thought it is conventional to start model names with small letter, is it so?
I will revert this change, if it is conventional to use small letters just tell me and I will make new changes.

Noisyboy-9 committed:
Change checking for auth middleware usage

in the user.controller.js file there was the intention to check
for the user being authenticated and then doing the rest of
the logic which was done wrong, the getUser() function has to be called
whcih returns the auth user.

Noisyboy-9 committed:
Changed user.controller.js to use user.service.js

Noisyboy-9 committed:
Merge remote-tracking branch 'origin/get_list_of_all_users' into get_all_tweets_of_user

# Conflicts:
#	src/controllers/user.controller.js

Noisyboy-9 committed:
Merge get_list_of_all_users into get_all_tweets_of_user

Noisyboy-9 committed:
Changed user.controller.js to use user.service.js

Noisyboy-9 committed:
Changed user.controller.js to use user.service.js for getting all tweets of user

Noisyboy-9 committed:
Fix problems is user.service.js

Noisyboy-9 committed:
Merge remote-tracking branch 'origin/get_list_of_all_users' into get_all_tweets_of_user

# Conflicts:
#	src/controllers/user.controller.js
#	src/services/user.service.js

Noisyboy-9 committed:
Merge get_list_of_all_users into get_all_tweets_of_user

arefehkompani commented on src/controllers/user.controller.js:
```diff
@@ -0,0 +1,47 @@
+
+        userService.getTweetsByStudentId(studentId)
+            .then(tweets => {
+                logger.info(`Get list of all tweets by student id ${studentId} successfully`);
```
`logger` is not defined

arefehkompani commented on src/db/queries/tweets/read.js:
```diff
@@ -0,0 +1,6 @@
+const Tweet = require('../../models/tweets');
+module.exports = {
+    async getTweetsByUserId(id) {
```
We don't need `async` here.

arefehkompani commented on src/db/queries/user/read.js:
```diff
@@ -0,0 +1,11 @@
+        return User.find({}, ['name', 'studentId', 'avatar']);
+    },
+
+    async getUserByStudentId(studentId) {
```
We don't need `async` here too.

Noisyboy-9 committed:
Fix problems in code reviews

Noisyboy-9 commented:
changes requested have been implemented, please take a look and give me another review
